src/include/btree.h

```diff
@@ -102,7 +102,8 @@ typedef enum { /* Start position for eviction walk */
 struct __wt_btree {
     WT_DATA_HANDLE *dhandle;
 
-    WT_CKPT *ckpt; /* Checkpoint information */
+    WT_CKPT *ckpt;         /* Checkpoint information */
+    size_t ckpt_allocated; /* Checkpoint information array allocation size */
```

Comment:
I'd suggest changing the name of `ckpt_allocated` to `ckpt_bytes_allocated`, I spent some time thinking it was a boolean that indicated we'd allocated WT_CKPT structures.